Reflected aria attributes containing element references (#38678)
mdn/content: files/en-us/web/accessibility/aria/reference/attributes/aria-controls/index.md

## Associated interfaces

- {{domxref("Element.ariaControlsElements")}} and {{domxref("ElementInternals.ariaControlsElements")}}

>   - : The `ariaControlsElements` property is part of each element's interface. Its value is an array of {{domxref("Element")}}s corresponding to the ID values specified in the `aria-controls` attribute (for cases where those ID values reference valid and in-scope elements).

Reviewer: Is it only me that finds usages of identifier names mixed into prose grating? Like "An array of `Element`s", rather than "An array of `Element` instances"?

>   - : The `ariaControlsElements` property is part of each element's interface. Its value is an array of {{domxref("Element")}}s corresponding to the ID values specified in the `aria-controls` attribute (for cases where those ID values reference valid and in-scope elements).

Author: No, I find it very grating, at least visually. 

> What does "valid" mean here? "valid and in-scope elements" implies that we have 2 conditions, but what would it look like for an element to be in scope but not valid?

Good point. Valid means "exists" - i.e. I can reference ID "fred" but if no element with `id="fred"`. I think I can just use valid to mean "must exist and be in scope". These will then become:

```suggestion
  - : The `ariaControlsElements` property is part of each element's interface. Its value is an array of {{domxref("Element")}} instances corresponding to the valid ID values specified in the `aria-controls` attribute.
```
